pkg/module/go_get_fetcher.go

```diff
@@ -34,38 +34,30 @@ func (g *goGetFetcher) Fetch(mod, ver string) (Ref, error) {
 	// setup the GOPATH
 	goPathRoot, err := afero.TempDir(g.fs, "", "athens")
 	if err != nil {
-		// TODO: return a ref for cleaning up the goPathRoot
-		// https://github.com/gomods/athens/issues/329
 		ref.Clear()
-		return ref, err
+		return newDiskRef(g.fs, goPathRoot, "", ""), err
```

Comment:
Can we safely return nil here, and remove the ref.Clear() one line above?

Reply:
If we return a nil here the caller will not be able to call .Clear() to delete the temp dir created on disk for the GOPATH (that's done in line 35). This applies to all the subsequent comments as well. 

Yes I believe we could remove the ref.Clear() above since this it's using a NoOp ref, but thought it made sense to call Clear once we're done with the ref in case there is actually some cleanup involved in the future. For now, yes we can definitely get rid of it!